src/node_crypto.cc

```diff
@@ -1749,6 +1751,20 @@ int SSLWrap<Base>::NewSessionCallback(SSL* s, SSL_SESSION* sess) {
 }
 
 
+template <class Base>
+void SSLWrap<Base>::KeylogCallback(const SSL* s, const char* line) {
+  Base* w = static_cast<Base*>(SSL_get_app_data(s));
+  Environment* env = w->ssl_env();
+  HandleScope handle_scope(env->isolate());
+  Context::Scope context_scope(env->context());
+
+  Local<String> line_str = String::NewFromUtf8(env->isolate(),
+      line, NewStringType::kNormal).ToLocalChecked();
+  Local<Value> argv[] = { line_str };
+  w->MakeCallback(env->onkeylog_string(), arraysize(argv), argv);
```

Comment:
Minor performance optimization: change `line_str` to `Local<Value>` and invoke the callback like this:
```c++
w->MakeCallback(env->onkeylog_string(), 1, &line_str);
```

Reply:
Done! Out of curiosity, I inspected the bytecode and it seems GCC was already making that optimization on my platform.